src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/DefaultAuthorizationServerMetadataResolver.kt

```diff
@@ -85,3 +78,23 @@ internal class DefaultAuthorizationServerMetadataResolver(
  */
 private fun CIAuthorizationServerMetadata.expectIssuer(expected: HttpsUrl) =
     require(issuer == Issuer(expected.value.toURI())) { "issuer does not match the expected value" }
+
+/**
+ * Gets a well-known metadata URL of an issuer.
+ *
+ * @receiver the Issuer
+ * @param metadata the well-known metadata URL to get
+ * @param specCompliant whether to get a spec-compliant URL or not
+ */
+private fun HttpsUrl.metadataUrl(metadata: String, specCompliant: Boolean = true): Url =
+    with(Url(this.value.toString())) {
```

Comment:
I can hardly follow the code and interpret `this` 
Consider using full syntax for `this` or even better use local val(s).

Also I don't get the use of the first with()

For example, looks more clean IMO

```kotlin
         URLBuilder(this).apply {
            when (lookup){
                Lookup.BySpecification ->
                    path("/${metadata.removePrefix("/").removeSuffix("/")}${pathSegments.joinToString("/")}")
                Lookup.CommonDeviation ->
                    appendPathSegments("/${metadata.removePrefix("/").removeSuffix("/")}", encodeSlash = false)
            }
        }.build()
```